lib/cache.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+module.exports = class Cache {
+  constructor() {
+    this.cache = {};
+  }
+
+  set(id, value) {
+    this.cache[id] = value;
+  }
+
+  has(id) {
+    return this.cache[id] != null;
```

Comment:
This line will intend:

```js
    return this.cache[id] !== undefined;
```

And also, maybe be better:

```js
    return this.cache.hasOwnProperty(cacheId);
```

The later works well with `delete` operator, too.

Reply:
https://github.com/hexojs/hexo/blob/fa59eb28e8344c84f7fa17d4ee48f8def79ce07e/lib/plugins/helper/fragment_cache.js#L10

`!= null` is used in fragment_cache helper and it will return `false` when meets `undefined`.

Reply:
https://repl.it/repls/PertinentEffectiveLicenses

I have setup a repl and find out `hasOwnProperty` is just too slow.

Reply:
> Unless we want to cache null value, otherwise I believe != null will work well.

I think there might be edge case where `null` value is intended to be cached. Checking for undefined is more accurate.

Reply:
@curbengh @seaoak 

I have updated to use `id in this.cache` which makes `undefined` & `null` both cacheable.